qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -377,18 +445,53 @@ def traces(self) -> ChannelList:
         """
         Update channel list with active traces and return the new list
         """
-        parlist = self.ask("CALC:PAR:CAT:EXT?").strip('"').split(",")
+        # Keep track of which trace was active before. This command may fail
+        # if no traces were selected.
+        try:
+            active_trace = self.active_trace()
+        except VisaIOError:
```

Comment:
Please check the contents of the exception message because if the reason for the exception is not that "no traces are selected", the driver might do smth wrong (or not do).